### PR TITLE
138 Fix setstate warning, turn on new eslint rule.

### LINKED ISSUE #138: Turn on eslint rule for "react/no-access-state-in-setstate"
Overview
===
This ticket covers the rule to add an eslint rule to ensure we are not making (obvious) error with the `setState` function. 

Requirements
===
The rule should be added to the `.eslintrc` file in the client folder. 
```json
"react/no-access-state-in-setstate": [2]
```

We then have to fix errors this rule picks up, currently there are 12 errors. 

Example code that needs to be refactored:
```js
  _formInput = (stateKey) => {
    return (e) => {
      const { store } = this.state;

      _.set(store, stateKey, e.target.value);
      this.setState({ ...this.state, store });
    };
  };
```

### PR DESCRIPTION
**Pull request by mikelax**
Closes #138 

- [X] Make fixes to `setState` calls. Notice most of the use cases required the same fix. I don't think there is a reason to use `...this.state`
- [X] fixed bug on Settings page where the data wasn't loading when transitioning from another page. It worked on fresh pageload. Fixed by updating to 16.3 lifecycle events.

### CODE REVIEW

File: client/src/components/Settings/index.js [hunk 1]
```diff
@@ -16,7 +16,16 @@ import { getProfile } from '../../services/webAuth';
...
-  state = { 
+  constructor() {
+    super();
+
+    getProfile()
+      .then(profile => {
+        this.setState({ profile });
+      });
+  }
+
```
Comment (nazar) [inline client/src/components/Settings/index.js]: I'm wondering if the `getProfile()` block should go in `componentDidMount` as I think calling async functions in a constructor is discouraged.

Plus it saves is from calling `super()` which can have gotchas if `this.props` ever need to be accessed in the constructor.

File: client/src/components/Settings/index.js [hunk 2]
```diff
@@ -30,22 +39,15 @@ class Settings extends Component {
...
-  componentWillMount() {
-    return getProfile()
-      .then(profile => {
-        this.setState({ profile });
-      });
-  }
-
-  componentWillReceiveProps(nextProps) {
-    this.setState({
+  static getDerivedStateFromProps(props) {
```
Comment (nazar) [inline client/src/components/Settings/index.js]: NICE! 👍

File: client/src/components/Settings/index.js (hunk 1 above)
Comment (mikelax) [reply client/src/components/Settings/index.js]: @nazar  ah, well i did see that i should be passing `props` in the super. 
but i will give it a shot. 

In [the docs](https://reactjs.org/docs/react-component.html#constructor), are you referring to this sentence with regards to having async functions in constructor?
>Avoid introducing any side-effects or subscriptions in the constructor. For those use cases, use componentDidMount() instead.

**Review by nazar — APPROVED** [review]
🚀 🚢

**Pull request merged**